Sources/Controllers/TargetMenu/OAEditPointViewController.mm

```diff
@@ -490,26 +490,8 @@ - (void)createIconList
 
 - (void)setupColors
 {
-    UIColor *loadedColor = [_pointHandler getColor];
-
     _appearanceCollection = [OAGPXAppearanceCollection sharedInstance];
-    if (_editPointType == EOAEditPointTypeFavorite)
-    {
-        _selectedColorItem =
-            [_appearanceCollection getColorForFavoriteGroupName:_favorite ? [_favorite getCategory]
-                                                                          : _isNewItemAdding ? [[OAAppSettings sharedManager].lastFavCategoryEntered get] : @""
-                                                      pointName:self.name
-                                                   defaultValue:[OAUtilities colorToNumberFromString:[loadedColor toHexARGBString]]];
-    }
-    else
-    {
-        _selectedColorItem = [_appearanceCollection getColorForGpxFilePath:_waypoint ? _waypoint.docPath : _gpxFileName
-                                                                 groupName:_waypoint ? _waypoint.point.type : @""
-                                                                 pointName:self.name
-                                                              defaultValue:[OAUtilities colorToNumberFromString:[loadedColor toHexARGBString]]];
-    }
-    if (!_selectedColorItem)
-        _selectedColorItem = [_appearanceCollection getDefaultPointColorItem];
+    _selectedColorItem = [_appearanceCollection getColorItemWithValue:[OAUtilities colorToNumberFromString:[[_pointHandler getColor] toHexARGBString]]];
```

Comment:
Extra double conversion. Introduce `toARGB` method to `UIColor (util)` (OAUtilities)